docs/src/main/sphinx/connector/pinot.rst

```diff
@@ -74,6 +74,11 @@ Property name                                             Required   Description
 ``pinot.aggregation-pushdown.enabled``                    No         Push down aggregation queries, default is ``true``.
 ``pinot.count-distinct-pushdown.enabled``                 No         Push down count distinct queries to Pinot, default is ``true``.
 ``pinot.target-segment-page-size``                        No         Max allowed page size for segment query, default is ``1MB``.
+``pinot.extra-http-headers``                              No         Extra headers when sending HTTP based pinot requests to Pinot controller/broker.
+                                                                     Headers are comma-separated, and each header key value pair is colon-separated,
+                                                                     e.g. ``k1:v1,k2:v2,k3:v3 with space`` with space, default is an empty map. Note
```

Comment:
A few questions:
* What happens when keys and values contain characters that are not valid characters in the HTTP message header syntax?
* What happens if the user provides headers that conflict with standard HTTP headers (e.g., Content-Type, Accept, etc)
* What's the backward compatibility story if the connector starts sending a header in a new version that the user had configured manually?

Reply:
1. Technically user can send any character;
2. Extra headers are set first, then standard HTTP headers. See method: `doHttpActionWithHeadersJson()`;
3. We call it user-side backward incompatible. Typically new features added may require a flag to turn on, so users should get a client-side warning or just an exception up to the feature implementor.